bundles/org.openhab.binding.yioremote/src/main/java/org/openhab/binding/yioremote/internal/YIOremoteDockHandler.java

```diff
@@ -0,0 +1,380 @@
+        switch (yioRemoteDockActualStatus) {
+            case CONNECTION_ESTABLISHED:
+                authenticationmessagehandler.setToken(localConfig.accessToken);
+                sendMessage(YioRemoteMessages.AUTHENTICATE_MESSAGE, localConfig.accessToken);
+                yioRemoteDockActualStatus = YioRemoteDockHandleStatus.AUTHENTICATION_PROCESS;
+                break;
+            case AUTHENTICATION_PROCESS:
+                if (authenticationok) {
+                    yioRemoteDockActualStatus = YioRemoteDockHandleStatus.AUTHENTICATION_COMPLETE;
+                    updateStatus(ThingStatus.ONLINE);
+                    websocketpollingjob = scheduler.scheduleWithFixedDelay(this::pollingWebsocket, 0, 30,
+                            TimeUnit.SECONDS);
+                    if (authenticationjob != null) {
+                        authenticationjob.cancel(true);
+                    }
+                } else {
+                    yioRemoteDockActualStatus = YioRemoteDockHandleStatus.AUTHENTICATION_FAILED;
+                }
+                break;
+            default:
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Connection lost no ping from YIO DOCK");
+                updateState(GROUP_OUTPUT, STATUS_STRING_CHANNEL, UnDefType.UNDEF);
+                break;
+        }
+    }
+
+    private void pollingWebsocket() {
+        if (yioRemoteDockActualStatus.equals(YioRemoteDockHandleStatus.AUTHENTICATION_COMPLETE)) {
+            if (getAndResetHeartbeat()) {
+                updateChannelString(GROUP_OUTPUT, STATUS_STRING_CHANNEL,
+                        ircodereceivedhandler.getCode() + ircodereceivedhandler.getFormat());
+                logger.debug("heartbeat ok");
+                sendMessage(YioRemoteMessages.HEARTBEAT_MESSAGE, "");
+            } else {
+                yioRemoteDockActualStatus = YioRemoteDockHandleStatus.CONNECTION_FAILED;
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Connection lost no ping from YIO DOCK");
+                updateState(GROUP_OUTPUT, STATUS_STRING_CHANNEL, UnDefType.UNDEF);
+                if (websocketpollingjob != null) {
+                    websocketpollingjob.cancel(true);
+                }
+            }
+        } else {
+            if (websocketpollingjob != null) {
+                websocketpollingjob.cancel(true);
+            }
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Connection lost no ping from YIO DOCK");
+            updateState(GROUP_OUTPUT, STATUS_STRING_CHANNEL, UnDefType.UNDEF);
+        }
+    }
+
+    public boolean getAndResetHeartbeat() {
+        boolean result = heartbeat;
+        heartbeat = false;
+        return result;
+    }
+
+    public YioRemoteDockHandleStatus getyioRemoteDockActualStatus() {
+        return yioRemoteDockActualStatus;
+    }
```

Comment:
Can you access the field directly without using this method?

Reply:
I learned thatis always better to have a get function instead of accessing directly. i use this also in the action handler therefore i have this with get